src/libAtomVM/bitstring.c

```diff
@@ -119,6 +135,56 @@ bool bitstring_utf8_encode(avm_int_t c, uint8_t *buf, size_t *out_size)
     return true;
 }
 
+bool bitstring_utf8_decode(int32_t *c, uint8_t *buf, size_t len, size_t *out_size)
+{
+    *c = 0;
```

Comment:
Do we really want to clear *c if the input is invalid?

Reply:
I guess I take the opinion here is that all bets are off if you get a `false` return value from the function, for any of the reference params.  Basically, you are calling a function with side effects, so buyer beware.

Reply:
Fixed